### PR TITLE
@:inheritDoc(pack.MyType.field)

### PR DESCRIPTION
**Pull request by RealyUniqueName**
Adds `@:inheritDoc` meta to append documentation from a parent field or class (if used without an argument) or from a specified field (if used like @:inheritDoc(pack.Some.field)).
Haxe compiler wil automatically append  the doc on display requests.

### CODE REVIEW

**Comment by RealyUniqueName** [comment]
@Simn I've pushed all the noise separately in https://github.com/HaxeFoundation/haxe/commit/279f459a5b12aa074198acad59b477b32c542c23
You can check this PR now :)

File: src/typing/typeloadFields.ml [hunk 1]
```diff
@@ -1032,6 +1032,77 @@ let check_abstract (ctx,cctx,fctx) c cf fd t ret p =
...
+				| _ -> raise Exit
+			with Exit | Invalid_argument _ ->
+				error "Invalid argument for @:inheritDoc" p
+		in
+		let t =
+			let tp = { tpackage = pack; tname = name; tsub = sub; tparams = []; } in
+			try Some (Typeload.load_instance ctx (tp,p) true)
+			with Error (Module_not_found _,p2) when p == p2 -> None
+		in
+		flush_pass ctx PBuildClass "inheritDoc";
```
Comment (Simn) [inline src/typing/typeloadFields.ml]: Why do we have to do this here?

**Comment by Simn** [comment]
I'm not sure if `doc_inherited: (unit -> (string option)) list` is a good approach. I would prefer to not have closures in the syntactic representation...

File: src/typing/typeloadFields.ml (hunk 1 above)
Comment (RealyUniqueName) [reply src/typing/typeloadFields.ml]: Because we need class structure to be ready at this point to be able to retrieve docs of class fields.

Comment (RealyUniqueName) [reply src/typing/typeloadFields.ml]: Oh, right. We don't have to, because actual doc retrieval is delayd to `PConnectField` anyway.

**Comment by RealyUniqueName** [comment]
> I'm not sure if `doc_inherited: (unit -> (string option)) list` is a good approach. I would prefer to not have closures in the syntactic representation...

Changed to `ref` values instead of closures.
But what bothers me in this case is that class loading order may affect the doc inheritance.

**Comment by Simn** [comment]
> Changed to ref values instead of closures.
> But what bothers me in this case is that class loading order may affect the doc inheritance.

Why do we need `doc_inherited` at all? To save computations? I'd argue that we could handle this sort of caching while preparing the output. Just stuff the computed docs into some Hashtbl.

**Comment by Simn** [comment]
@RealyUniqueName Please update the branch and check my last comment.

**Comment by RealyUniqueName** [comment]
AFAIR I did start to refactor out `doc_inherited`, but got stuck at some point. I'll get back to this PR sometime later, when I'll have my own PR "workout".

**Comment by nadako** [comment]
@RealyUniqueName do you think this can make it into 4.1?

**Comment by RealyUniqueName** [comment]
Nah

**Comment by RealyUniqueName** [comment]
Closing in favor of #9817

**Pull request closed without merging**